src/main/java/teammates/common/exception/UnauthorizedAccessException.java

```diff
@@ -6,8 +6,34 @@
 @SuppressWarnings("serial")
 public class UnauthorizedAccessException extends RuntimeException {
 
+    /**
+     * Indicates whether the error message will be shown to the user (as the API response).
+     * If this flag is set to "false", a generic error message will be displayed instead.
+     * The actual error message is always displayed to admin in the logs.
+     *
+     * <p>As this exception involves access control, the error message need to be redacted
+     * in most cases because the full error message can give clues as to what needs to be done
+     * in order to get the access (which should be unauthorized).
+     * While it is true that there is little that user can do to elevate his/her privilege
+     * even with that knowledge, we still need to do our part.
+     *
+     * <p>There are, however, some cases whereby the error message would be legitimately helpful
+     * to users, e.g. when submitting responses to a feedback session that has not opened yet.
+     * For such cases, it is appropriate to set this flag as "true".
+     */
+    private final boolean showErrorMessage;
+
     public UnauthorizedAccessException(String message) {
+        this(message, false);
```

Comment:
Is there any specific reason why the default is false? I think it should be true since this is the behavior for all other error message types (whatever you write is displayed to the user). There's also a lot more cases of you passing in true as the 2nd parameter compared to using it as false, although this isn't really a compelling argument.

Reply:
> There's also a lot more cases of you passing in true as the 2nd parameter compared to using it as false

I'm not sure where you get this from, because only 5 out of 55 `UnauthorizedAccessException` being created have true as the 2nd param.

Reply:
> I'm not sure where you get this from, because only 5 out of 55 `UnauthorizedAccessException` being created have true as the 2nd param.

Ah ok, I only looked though the changes so I just found more cases of it being true than not. (I realize this is on a very biased sample)

Dropping that, what about keeping it to default false for consistency with other exception types?

Reply:
It's not about consistency. If you understand the reason why it needs to be redacted in the first place, you should know why the default is false. Admittedly, even that reason is debatable.